src/goto-symex/symex_stack.cpp

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************\
+
+Module: Symbolic Execution - Stack tracking
+
+Author: Rafael Sá Menezes
+
+Date: April 2020
+
+\*******************************************************************/
+
+#include <cassert>
+#include <goto-symex/goto_symex.h>
+#include <util/expr_util.h>
+#include <util/irep2.h>
+
+void goto_symext::process_stack_size(expr2tc &expr)
+{
+  code_decl2t &decl_code = to_code_decl2t(expr);
+  cur_state->top().frame_info.add_decl(decl_code);
+  if(stack_limit != -1)
```

Comment:
Why "-1"?

Reply:
The value of `stack_limit`comes from an `atol` and since the default value for the parameter is -1 this checks if the user has changed it.

A better approach could be done when using BigInt, but we have plans of replacing it: https://github.com/esbmc/esbmc/issues/181